### ISSUE TITLE
Incorrect printing of `StepRangeLen{CartesianIndex{2}, CartesianIndex{2}, CartesianIndex{2}}`

### ISSUE THREAD

**Issue by LilithHafner** [JuliaLang/julia#50784]
`CartesianIndex(1, 2):CartesianIndex(3, 4):CartesianIndex(4, 6)` means something really weird, so we can't use it in `show`.

```julia
julia> StepRangeLen(CartesianIndex(1,2), CartesianIndex(3,4), 2)
CartesianIndex(1, 2):CartesianIndex(3, 4):CartesianIndex(4, 6)

julia> CartesianIndex(1, 2):CartesianIndex(3, 4):CartesianIndex(4, 6)
2×2 CartesianIndices{2, Tuple{StepRange{Int64, Int64}, StepRange{Int64, Int64}}}:
 CartesianIndex(1, 2)  CartesianIndex(1, 6)
 CartesianIndex(4, 2)  CartesianIndex(4, 6)
```

**Comment by gbaraldi** [comment]
Specifically the colon constructor does something unexpected because it doesn't respect that ranges should be constructed by repeateadly adding the step.

**Comment by LilithHafner** [comment]
Fixed by #50457
